sdk/js/src/relayer/__tests__/wormhole_relayer.ts

```diff
@@ -0,0 +1,690 @@
+      ethers.utils.toUtf8Bytes(generateRandomString(32))
+    );
+    console.log(`Sent message: ${arbitraryPayload1}`);
+    
+    const txHash = await testForward(arbitraryPayload1, arbitraryPayload2);
+
+    await waitForRelay(2);
+
+    console.log("Checking if message was relayed");
+    const message1 = await targetMockIntegration.getMessage();
+    console.log(
+      `Sent message: ${arbitraryPayload1} (expecting ${arbitraryPayload2} from forward)`
+    );
+    console.log(`Received message on target: ${message1}`);
+    expect(message1).toBe(arbitraryPayload1);
+
+    console.log("Checking if forward message was relayed back");
+    const message2 = await sourceMockIntegration.getMessage();
+    console.log(`Sent message: ${arbitraryPayload2}`);
+    console.log(`Received message on source: ${message2}`);
+    expect(message2).toBe(arbitraryPayload2);
+
+    const status = await getStatus(txHash);
+    expect(status).toBe("Forward Request Success");
+  });
+
+  test("Executes a multidelivery", async () => {
+    const arbitraryPayload1 = ethers.utils.hexlify(
+      ethers.utils.toUtf8Bytes(generateRandomString(32))
+    );
+    const arbitraryPayload2 = ethers.utils.hexlify(
+        ethers.utils.toUtf8Bytes(generateRandomString(32))
+      );
+    console.log(`Sent message: ${arbitraryPayload1}`);
+   
+    const txHash1 = await testSend(arbitraryPayload1, true);
+    const txHash2 = await testSend(arbitraryPayload2);
+
+    await waitForRelay();
+
+    console.log("Checking if first message was relayed");
+    const message1 = await sourceMockIntegration.getMessage();
+    console.log(`Sent message: ${arbitraryPayload1}`);
+    console.log(`Received message: ${message1}`);
+    expect(message1).toBe(arbitraryPayload1);
+
+    console.log("Checking if second message was relayed");
+    const message2 = await targetMockIntegration.getMessage();
+    console.log(`Sent message: ${arbitraryPayload2}`);
+    console.log(`Received message: ${message2}`);
+    expect(message2).toBe(arbitraryPayload2);
+  });
+
+  test("Executes a multiforward", async () => {
+    const arbitraryPayload1 = ethers.utils.hexlify(
+      ethers.utils.toUtf8Bytes(generateRandomString(32))
+    );
+    const arbitraryPayload2 = ethers.utils.hexlify(
+      ethers.utils.toUtf8Bytes(generateRandomString(32))
+    );
+    console.log(`Sent message: ${arbitraryPayload1}`);
+    const value1 = await sourceCoreRelayer.quoteGas(
+      targetChainId,
+      500000,
+      await sourceCoreRelayer.getDefaultRelayProvider()
+    );
+    const value2 = await targetCoreRelayer.quoteGas(
+      sourceChainId,
+      500000,
+      await targetCoreRelayer.getDefaultRelayProvider()
+    );
+    const value3 = await targetCoreRelayer.quoteGas(
+      targetChainId,
+      500000,
+      await targetCoreRelayer.getDefaultRelayProvider()
+    );
+    const payment = value1
+      .mul(2)
+      .add(value2)
+      .add(value3)
+      .mul(105)
+      .div(100)
```

Comment:
presumably that's the asset conversion buffer?

Reply:
Yes, added comment